### ISSUE TITLE
agGridReact.ts still references componentWillReceiveProps() --- See Plunker

### ISSUE THREAD

**Issue by sljidexcel** [ag-grid/ag-grid#3366]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

**Customers of ag-Grid**

If you are a customer you are entitled to use the ag-Grid's customer support system (powered by Zendesk). Please use that channel for guaranteed response from the ag-Grid team with regards bugs, feature requests and support.

**Requesting Community Support**

If you are not a customer of ag-Grid, ag-grid staff will label your issue as managed-by-the-community. This means that ag-Grid staff is not going to be actively looking into it and it will get closed if inactive for more than one month. The community is welcome to help with this question/support issue.

**Providing a Reproducible Scenario**
Plunker:

http://next.plnkr.co/edit/MSfA1biU4UzgssJ3?open=Hello.js&deferRun=1

When loaded, please click Preview and open the Plunker Console to see the warning generated.

**Current behavior**
With React/React DOM > 16.8.6 (Plunker uses ^16.9.0), a web console warning is issued encouraging you to update agGridReact because it's calling a deprecated lifecycle function.  However, the current version of agGridReact (21.1.0) has the offending code.

**Expected behavior**
Use a valid lifecycle hook in agGridReact to prevent the warning in React ^16.9.0

**Please tell us about your environment:**
MacOS, VSCode, NPM, Node.Js, React ^16.9.0, React DOM ^16.9.0

* **ag-Grid version:** X.X.X
21.1.0

* **Browser:**
Chrome Version 76.0.3809.132 (Official Build) (64-bit)

* **Language:**
ES6

**Comment by dmytsv** [comment]
Same. ag-grid-community@21.2.1, ag-grid-enterprise@21.2.1, ag-grid-react@21.2.1

**Comment by AhmedAGadir** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by AhmedAGadir** [comment]
Hi,

Thank you for your email. This is in our backlog now and we are tracking it with the following reference and description.

AG-3316 - agGridReact needs to be updated to use the updated react lifecycle hooks

You can follow the status of feature requests, bugs and releases using our pipeline: 

https://www.ag-grid.com/ag-grid-pipeline/

(Note that newly raised issues can take up to 2 hours to appear in the pipeline)

Regards

Ahmed‍

**Comment by sljidexcel** [comment]
Thank you so much for your help.  Respectfully, shade

**Comment by sljidexcel** [comment]
Hello,

On 9/11 I received an email stating this issue was being placed in a backlog:

AG-3316 - agGridReact needs to be updated to use the updated react lifecycle hooks
You can follow the status of feature requests, bugs and releases using our pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

I've never gotten an update.  When I visit the pipeline, the search on AG-3316 yields nothing.

I do not understand how this ticket is closed when, looking at agGridReact in Github, the latest code seems to still have the deprecated function.  Furthermore, given the problems with React 16.8.6:

https://github.com/facebook/react/issues/15970

The actual resolution of this ticket is really important.

Somebody, please provide some insight.  Thank you | shade

**Comment by sljidexcel** [comment]
My apologies.  I went to the changelog:

https://www.ag-grid.com/ag-grid-changelog/

And saw that a fix to AG-3316 was included in version  22.0.0.

Thank you very much for your help.  Respectfully | shade
